src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -130,7 +131,7 @@ public void save(
             }
         }
 
-        raiseAppInsightEvents(updatedClaim, response, claimantResponse);
+        raiseAppInsightEvents(updatedClaim, response, claimantResponse, caseEvent.orElseGet(() -> null));
```

Comment:
No need to lazy retrieve a null

```suggestion
        raiseAppInsightEvents(updatedClaim, response, claimantResponse, caseEvent.orElse(null));
```